File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<Map, List> resMap = new HashMap<>();
        //foreach string : strs, calculate freqmap
        for (String str : strs) {
            Map<Character, Integer> freq = new HashMap<>();
            for (Character c : str.toCharArray()) {
                freq.put(c, freq.getOrDefault(c, 0) + 1);
            }
            List<String> ls = resMap.getOrDefault(freq, new ArrayList<>());
            ls.add(str);
            resMap.putIfAbsent(freq, ls);
        }
        List<List<String>> res = new ArrayList<>();
        resMap.values().stream().forEach(ls -> res.add(ls));
        return res;
        //if freqmap in key: put in list, else create entry
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def getFreqMap(self, s) -> dict:
        freq_map = {}
        for c in s:
            freq_map[c] = freq_map.get(c, 0) + 1
        return freq_map

    def isAnagram(self, s: str, t: str) -> bool:
        return self.getFreqMap(s) == self.getFreqMap(t)